core/src/main/java/de/jplag/highlightextraction/FrequencyAnalysisOptions.java

```diff
@@ -0,0 +1,58 @@
+package de.jplag.highlightextraction;
+
+/**
+ * Options for Frequency Analysis.
+ * @param isFrequencyAnalysisEnabled if isFrequencyAnalysisEnabled analysis is used
+ * @param frequencyStrategy the strategy used to determine the isFrequencyAnalysisEnabled of a Match
+ * @param frequencyStrategyMinValue the minimum considered size of Subsequences from matches in the frequencyStrategy
+ * @param weightingStrategy strategy used to influence the similarity based on Match isFrequencyAnalysisEnabled
+ * @param weightingFactor how strong the impact of the weightingStrategy is
+ */
+public record FrequencyAnalysisOptions(boolean isFrequencyAnalysisEnabled, FrequencyAnalysisStrategy frequencyStrategy, int frequencyStrategyMinValue,
```

Comment:
Not for this PR. But maybe for a follow-up. You can create builders like here: https://github.com/ardoco/lissa-ratlr/blob/f2db92ec2b079a9001782e9cd02cfcbf86bc0eac/src/main/java/edu/kit/kastel/sdq/lissa/ratlr/configuration/Configuration.java#L98